models/servers.go

```diff
@@ -0,0 +1,100 @@
+package models
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// Server maps AsyncAPI serves.item object
+type Server struct {
+	Extensions    Extensions       `json:"-"`
+	URL           string           `json:"url"`
+	Description   string           `json:"description,omitempty"`
+	Scheme        string           `json:"scheme"`
+	SchemeVersion string           `json:"schemeVersion,omitempty"`
+	Variables     *ServerVariables `json:"variables,omitempty"`
+	BaseChannel   string           `json:"baseChannel,omitempty"`
+	// Security      []*SecurityRequirement `json:"security,omitempty"`
+}
+
+// Unmarshal unmarshals JSON
+func (value *Server) Unmarshal(data []byte) error {
+	return json.Unmarshal(data, value)
```

Comment:
We should avoid calling `json.Unmarshal` inside the Unmarshal method. I know I've done it in the Info object but now we know there's the other way of doing it. This method should just create and return a new map and add property values one by one. If we do it and also rename the method to `UnmarshalJSON` the `json.Unmarshal` function will automatically call it.

Reply:
great I wanted to get your confirmation on this, 100% agree